identity/aziot-identityd/src/identity.rs

```diff
@@ -614,12 +616,25 @@ impl IdentityManager {
                         identity_cert,
                         identity_pk,
                     } => {
-                        self.create_identity_cert_if_not_exist_or_expired(
-                            &identity_pk,
-                            &identity_cert,
-                            &registration_id,
-                        )
-                        .await?;
+                        //DPS requires registration ID to match identity certificate CN
+                        let cert_subject_name = self
+                            .create_identity_cert_if_not_exist_or_expired(
+                                &identity_pk,
+                                &identity_cert,
+                                registration_id.as_deref().to_owned(),
+                            )
+                            .await?;
+
+                        let registration_id = match registration_id {
```

Comment:
This block is more concise with `unwrap_or_else`.

Reply:
The unwrap_or_else doesn't permit lazy evaluatation of result of create_identity_cert_if_not_exist_or_expired. This is intentional